File: app/src/main/java/wxm/KeepAccount/ui/data/show/note/ACDailyDetail.java
package wxm.KeepAccount.ui.data.show.note;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;

import wxm.androidutil.Switcher.ACSwitcherActivity;
import wxm.androidutil.util.UtilFun;
import wxm.KeepAccount.define.GlobalDef;

/**
 * day detail UI
 */
public class ACDailyDetail extends ACSwitcherActivity<FrgDailyDetail> {
    public final static String K_HOTDAY = "hotday";

    @Override
    protected void leaveActivity() {
        int ret_data = GlobalDef.INTRET_GIVEUP;

        Intent data = new Intent();
        setResult(ret_data, data);
        finish();
    }

    @Override
    protected void setupFragment(Bundle bundle) {
        Intent it = getIntent();
        assert null != it;
        String hot_day = it.getStringExtra(K_HOTDAY);
        if (UtilFun.StringIsNullOrEmpty(hot_day)) {
            Log.e(LOG_TAG, "调用intent缺少'K_HOTDAY'参数");
            return;
        }

        // for holder
        FrgDailyDetail fg = new FrgDailyDetail();
        Bundle bd = new Bundle();
        bd.putString(K_HOTDAY, hot_day);
        fg.setArguments(bd);

        addFragment(fg);
    }
}

File: app/src/main/java/wxm/KeepAccount/ui/data/show/note/ACDailyDetail.kt
package wxm.KeepAccount.ui.data.show.note

import android.os.Bundle
import wxm.KeepAccount.ui.base.ACBase.ACBase
import wxm.KeepAccount.ui.data.show.note.page.FrgDailyDetail

import wxm.androidutil.log.TagLog

/**
 * day detail UI
 */
class ACDailyDetail : ACBase<FrgDailyDetail>() {
    override fun setupFragment(): MutableList<FrgDailyDetail> {
        val hotDay = intent!!.getStringExtra(KEY_HOT_DAY)
        if (hotDay.isNullOrEmpty()) {
            TagLog.e("调用intent缺少'KEY_HOT_DAY'参数")
            return arrayListOf()
        }

        // for holder
        return arrayListOf(FrgDailyDetail().apply {
            arguments = Bundle().apply { putString(KEY_HOT_DAY, hotDay) }
        })
    }

    companion object {
        const val KEY_HOT_DAY = "hot_day"
    }
}
